pootle/apps/pootle_store/forms.py

```diff
@@ -280,11 +284,16 @@ def clean_target_f(self):
 
             return value
 
-        def clean_state(self):
+        def _clean_state(self):
+            # Handle clean_state here because it depends on clean_suggestion.
```

Comment:
@phlax I had to change method name because `clean_state()` depends on `clean_suggestion()` and `clean_fields()` doesn't assume any ordering of fields. That's why I call `_clean_state()` as real `clean_state()` but from `clean()` when there is `cleaned_data` for all fields. Do you follow?